benchexec/tools/ultimate.py

```diff
@@ -204,113 +204,116 @@ def _requires_ultimate_data(self, executable):
         # all versions before 0.1.24 do not require ultimatedata
         return not (int(major) == 0 and int(minor) < 2 and int(patch) < 24)
 
-    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits=None):
-        if rlimits is None:
-            rlimits = {}
+    def cmdline(self, executable, options, task, resource_limits):
+        self._uses_propertyfile = task.property_file is not None
```

Comment:
I did not notice this before, but storing information about a task in `self` is broken, because there is no guarantee that `cmdline` and `determine_result` are called in any particular order (cf. docs of `BaseTool2`). So please remove this field. With the new API this should finally be possible because `determine_result` has access to the run command line.